gbus/metrics/handler_metrics.go

```diff
@@ -37,29 +52,31 @@ func AddHandlerMetrics(handlerName string) {
 }
 
 //RunHandlerWithMetric runs a specific handler with metrics being collected and reported to prometheus
-func RunHandlerWithMetric(handleMessage func() error, handlerName string, logger logrus.FieldLogger) error {
+func RunHandlerWithMetric(handleMessage func() error, handlerName, messageType string, logger logrus.FieldLogger) error {
 	handlerMetrics := GetHandlerMetrics(handlerName)
 	defer func() {
 		if p := recover(); p != nil {
 			if handlerMetrics != nil {
 				handlerMetrics.result.WithLabelValues(failure).Inc()
 			}
-
+			handlersResultCounter.WithLabelValues(messageType, failure).Inc()
```

Comment:
@danielwitz if a specific handler is registered on a specific message type then why do we need to report success/failures according to message type as well?
In addition, if I have two different handlers registered to the same message type wouldn't this value get incremented twice?

Reply:
@rhinof I was thinking of having this metric be with the handler's and message type as labels and then then user can choose on what to build their dashboard.
I think it would be important for handler like the deadletter queue ones.
will update my PR soon